spring-amqp/src/main/java/org/springframework/amqp/support/converter/AbstractJackson2MessageConverter.java

```diff
@@ -265,9 +291,14 @@ protected Message createMessage(Object objectToConvert, MessageProperties messag
 
 		byte[] bytes;
 		try {
-			String jsonString = this.objectMapper
-					.writeValueAsString(objectToConvert);
-			bytes = jsonString.getBytes(getDefaultCharset());
+			if (getDefaultCharset().equals("UTF-8")) {
```

Comment:
There is `StandardCharsets.UTF_8.UTF_8` to consider instead.

Although it even might be better to have that `defaultCharset` property as a `Charset` instance instead to fail early if provided string representation is not supported.

Reply:
Yeah - I just copied this from Ollie's code in Kafka,